### PR TITLE
Update `protoc-gen-go-flags` version

### LINKED ISSUE #5918: Update to version `1.0.6` of `protoc-gen-go-flags`
### Summary

Update the mage tool and the package dependency to the latest version. 

### Current Situation

_No response_

### Why do we need this? Who uses it, and when?

This version offers stable sorting of help flags which in turn makes our docs more stable.

### Proposed Implementation

_No response_

### Contributing

- [X] I can help by doing more research.
- [X] I can help by implementing the feature after the proposal above is approved.
- [X] I can help by testing the feature before it's released.

### Code of Conduct

- [X] I agree to follow TTN's [Community Code of Conduct](https://github.com/TheThingsNetwork/Manifesto/blob/master/CodeOfConduct.md).

### PR DESCRIPTION
**Pull request by ysmilda**
#### Summary

Closes #5918 

#### Changes

- Update `protoc-gen-go-flags` version to `1.0.6`

#### Testing

Manual testing.

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
- [x] Documentation: Relevant documentation is added or updated.
- [x] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

**Review by adriansmares — APPROVED** [review]

**Comment by adriansmares** [comment]
What is the status here ?

**Comment by ysmilda** [comment]
Waiting for approval from @johanstokking so it can be merged.

**Review by johanstokking — APPROVED** [review]
Apologies again, same issue

**Pull request merged**
